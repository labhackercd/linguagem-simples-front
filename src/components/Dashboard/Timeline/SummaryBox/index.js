import React, {useState} from 'react';
import { Grid, Typography, Box, Button, TextField } from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import axiosInstance from '../../../../auth/axiosApi.js';
import DownArrowIcon from './../../../../assets/down-arrow.svg';
import {API_SESSIONS_URL} from './../../../../api_urls'

const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 0.5rem 0',
  },
	notchedOutline: {
		border: '2px solid #F2F2F2',
		borderWidthBottom: '0px',
		color: theme.palette.secondary,
		borderRadius: '5px 5px 0 0',
	},
	textArea: {
		height: '100%',
		margin: '0rem 0 0 0',
		width: '100%',
		border: '0',
	},
	button: {
	  height: 'inherit',
	  alignSelf: 'center',
		borderRadius: '0 0 5px 5px',
		backgroundColor: theme.palette.verdeCamaraLight.main,
		color: theme.palette.white.main,
		textTransform: 'capitalize',
	},
	textField: {
		margin: '0 1rem 0 0',
		height: '100%',
		width: '100%',
	},
	box: {
		borderRadius: '0 0 0 5px',
		width: '100%',
	},
	buttonContainer: {
		display: 'flex',
		justifyContent: 'flex-end',
		 height: 'inherit'
	},
	container: {
		height: '100%',
	},
}));

export default function SummaryBox(props) {
  const classes = useStyles();
  const sessionID = props.sessionID;
  const [textFieldValue, setTextFieldValue] = useState('');
  function handleChange(e) {
    setTextFieldValue(e.target.value)
  }
  function handleSubmit() {
    const url = API_SESSIONS_URL + sessionID + '/';
    axiosInstance.patch(url, {
            id: sessionID,
            resume: textFieldValue
        }).then(
            result => {
              if(result.status === 200) {
                alert('Resumo atualizado com sucesso!')
              }
            }
    ).catch (error => {
        console.log(error)
    })

  }
  return (
    <React.Fragment>
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Resumo </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
    </Grid>
    <Grid item md={12}>
        <Grid container >
            <form className={classes.textArea} noValidate autoComplete="off">
              <TextField
                id="summary-box-textfield"
                multiline
                rows={4}
                value={textFieldValue}
                onChange={handleChange}
                variant="outlined"
								disabled={!props.broadcastingOnline}
                className={classes.textField}
                bgcolor="white"
                InputProps={{
                  classes: {
                    notchedOutline: classes.notchedOutline
                  },
                }}
              />
            </form>
          <Box className={classes.box} color="#F2F2F2" bgcolor="#F2F2F2">
						<Grid container className={classes.container}>
								<Grid item xs={8}></Grid>
								<Grid item xs={4} className={classes.buttonContainer}>
									<Button disabled={!props.broadcastingOnline} id="summary-box-submit-button" className={classes.button} onClick={(e) => handleSubmit(e)} variant="contained" disableElevation>
										Atualizar
									</Button>
								</Grid>
						</Grid>
          </Box>
      </Grid>
		</Grid>
    </React.Fragment>
  )
}
