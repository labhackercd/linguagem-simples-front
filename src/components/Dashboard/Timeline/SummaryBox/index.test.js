import React from 'react';
import { render, screen } from '@testing-library/react';
import SummaryBox from './index';
import {shallow} from "enzyme/build";

it("should render the summary box section", () => {
    const component = shallow(<SummaryBox/>);
    expect(component).toMatchSnapshot();
});
