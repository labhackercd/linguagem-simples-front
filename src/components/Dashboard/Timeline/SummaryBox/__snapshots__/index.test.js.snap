// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have differences 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-summaryBox-1"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-summaryHeader-2"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        style={
          Object {
            "display": "flex",
            "justifyContent": "flex-start",
          }
        }
      >
        <WithStyles(ForwardRef(Typography))
          variant="h5"
        >
           Resumo 
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        style={
          Object {
            "display": "flex",
            "justifyContent": "flex-end",
          }
        }
      >
        <a
          href="/"
        >
          <img
            alt="up arrow icon"
            src="down-arrow.svg"
          />
        </a>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <form
        autoComplete="off"
        className="makeStyles-textArea-4"
        noValidate={true}
      >
        <WithStyles(ForwardRef(TextField))
          InputProps={
            Object {
              "classes": Object {
                "notchedOutline": "makeStyles-notchedOutline-3",
              },
            }
          }
          bgcolor="white"
          className="makeStyles-textField-5"
          id="outlined-multiline-static"
          multiline={true}
          onChange={[Function]}
          rows={4}
          value=""
          variant="outlined"
        />
      </form>
    </WithStyles(ForwardRef(Grid))>
    <Styled(MuiBox)
      bgcolor="#F2F2F2"
      borderRadius="0 0 10px 25px"
      borderTop={1}
      color="#F2F2F2"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-start",
            }
          }
          xs={8}
        />
        <WithStyles(ForwardRef(Grid))
          item={true}
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
          xs={4}
        >
          <WithStyles(ForwardRef(Button))
            disableElevation={true}
            onClick={[Function]}
            variant="contained"
          >
            Atualizar
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
